### PR TITLE
add toLowerCamelCase function + unit tests

### PR DESCRIPTION
**Pull request by charlesakalugwu**
In preparation for json struct tags conformance tests... see #386

### CODE REVIEW

File: pkg/api/config_test.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+	for _, test := range testcases {
+		got := padNumbers(test.input)
+		if got != test.expected {
+			t.Errorf("padNumbers(%v) != %v, expected %v", test.input, got, test.expected)
+		}
+	}
+}
+
+// Converts a string to lower camel case
```
Comment (asalkeld) [inline pkg/api/config_test.go]: wouldn't it be a good idea to move the functional code out into a separate file from the start?

File: pkg/api/config_test.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	prep := s
+	start, rest := s[0], s[1:]
+	if start >= 'A' && start <= 'Z' {
+		prep = strings.ToLower(string(start)) + rest
+	}
+	//  pad all numbers within the string with spaces
+	prep = padNumbers(prep)
+
+	// trim spaces at the edges of the string
+	prep = strings.Trim(prep, " ")
```
Comment (asalkeld) [inline pkg/api/config_test.go]: shouldn't this be a bit up on line 47? to catch the case " ABook"

File: pkg/api/config_test.go [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+	return res
+}
+
+func TestLowerCamelCase(t *testing.T) {
+
+	testcases := []struct {
+		input    string
+		expected string
+	}{
```
Comment (asalkeld) [inline pkg/api/config_test.go]: maybe add some test cases that start with  a space

**Comment by kargakis** [comment]
> In preparation for json struct tags conformance tests... see #386

It's not clear from that issue what's the purpose of this PR. Maybe it would be clearer to try and use the functions you are unit-testing in prod code.

/hold
/cc jim-minter

File: pkg/api/config_test.go (hunk 1 above)
Comment (charlesakalugwu) [reply pkg/api/config_test.go]: this function will only be used from another testcase which will be within this file. That is why this is currently just in config_test.go. already discussed this with @jim-minter

File: pkg/api/config_test.go (hunk 3 above)
Comment (charlesakalugwu) [reply pkg/api/config_test.go]: done :)

File: pkg/api/config_test.go (hunk 2 above)
Comment (charlesakalugwu) [reply pkg/api/config_test.go]: good catch! thanks. added more testcases for this condition.

**Comment by charlesakalugwu** [comment]
@kargakis we would like to be able to walk through arbitrary structs in our project and verify that the json struct tags added to them conform to a standard that we set. That standard in our case is currently that all struct tags be set to the lower camel case format of the corresponding struct field. 

I am working on the next step, a function `jstCheck(o interface{}, f func(s string) string) []error` where f will be the `lowerCamelCase(string) string` function created in this pull request.

**Comment by charlesakalugwu** [comment]
@asalkeld the work is much more than just having a lower camel case function. Please see #505. I will close this PR as the more complete solution is in #505

**Pull request closed without merging**